### ISSUE TITLE
Multiprocessing.Queue causing abnormal behaviour 

### ISSUE THREAD

**Issue by pahansen95** [touilleMan/godot-python#237]
**OS**: OSX v10.15.6
**Godot**: v3.2.2
**godot-python**: v0.40.0

I'm using a python script attached to a node that imports a class I created from another python script & then instantiates the class in `_ready`.

This class in its `__init__` method creates an instance of a `Multiprocessing.Queue`.

When you Play the project Godot then proceeds to open hundreds of windows...

Not really sure what's happening; I'm hoping someone might have an idea.

`multiprocessing.Pipe` & `multiprocessing.Process` don't cause any issues.

I've attached an [Example Project](https://github.com/touilleMan/godot-python/files/5235500/multiprocessing_queue.zip) to replicate **BUT DON'T ATTEMPT TO RUN IT UNLESS YOU WANT TO CRASH YOUR COMPUTER**. I omitted the `godot-python` asset so you'll need to install that first.

**Comment by touilleMan** [comment]
Hi @pahansen95 !

That's really interesting, multiprocessing works by forking the current process so given Python is embedded into Godot it's very possible Godot gets pretty upset 😄 

I've tried your example on Windows, I didn't get your catastrophic crash but that's not surprising given multiprocessing is strongly tied to the underlying OS.

In the end I'm not sure what we can do except adding some information in the FAQ...

**Comment by pahansen95** [comment]
**TL;DR** It's a MacOS Thing; as a workaround call `multiprocessing.set_start_method('fork')` once per project but this is potentially unstable due to how MacOS implements fork.

Thanks for the response; based off what you said I dug through the CPython Source Code a bit & did some googling. It seems that MacOS Mojave changed how `fork` works which subsequently caused unexpected behaviour in Python for anything that used `fork`. In response `spawn` is now the default option for multiprocessing on MacOS since Python 3.8. Actually if I had read the [documentation fully](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods) I would of seen that.

My workaround at the moment is to call `multiprocessing.set_start_method('fork')`. This prevents Godot from opening endless windows & is executing my scripts but all the [issues & bug reports](https://www.google.com/search?q=python+macos+fork&oq=python+&aqs=chrome.0.69i59l4j69i60l4.7499j0j7&sourceid=chrome&ie=UTF-8) I read point to `fork` causing instability & coredumps so I don't think this can be considered a solution.

I'll follow up if I run into instabillities.

**Comment by pahansen95** [comment]
As promised here is my follow-up:

Switching multiprocessing from `spawn` to `fork` does indeed seem to crash both Godot & Standalone Python scripts whenever you attempt to terminate a process.
